crates/typos-dict/tests/verify.rs

```diff
@@ -9,38 +9,71 @@ type Dict = BTreeMap<UniCase<String>, BTreeSet<String>>;
 #[test]
 fn verify() {
     let asset_path = "assets/words.csv";
-    let data = std::fs::read(asset_path).unwrap();
+    let typos_dict = parse_dict(asset_path);
+    let new_dict = process(typos_dict);
 
     let mut content = vec![];
-    generate(&mut content, &data);
+
+    let mut wtr = csv::WriterBuilder::new()
+        .flexible(true)
+        .from_writer(&mut content);
+    for (typo, corrections) in new_dict {
+        let mut row = vec![typo.as_str().to_owned()];
+        row.extend(corrections);
+        wtr.write_record(&row).unwrap();
+    }
+    wtr.flush().unwrap();
+    drop(wtr);
 
     let content = String::from_utf8(content).unwrap();
     snapbox::assert_eq_path(asset_path, content);
 }
 
-fn generate<W: std::io::Write>(file: &mut W, dict: &[u8]) {
-    let mut rows = Dict::new();
-    csv::ReaderBuilder::new()
+fn parse_dict(path: &str) -> Dict {
+    let data = std::fs::read(path).unwrap();
+
+    let mut reader = csv::ReaderBuilder::new()
         .has_headers(false)
         .flexible(true)
-        .from_reader(dict)
+        .from_reader(&*data);
+
+    let iter = reader
         .records()
+        .into_iter()
         .map(Result::unwrap)
-        .for_each(|r| {
-            let mut i = r.iter();
-            let mut typo = i.next().expect("typo").to_owned();
-            typo.make_ascii_lowercase();
-            let typo = UniCase::new(typo);
-            rows.entry(typo)
-                .or_insert_with(BTreeSet::new)
-                .extend(i.map(|c| {
-                    let mut c = c.to_owned();
-                    c.make_ascii_lowercase();
-                    c
-                }));
+        .map(|record| {
+            let mut iter = record.into_iter();
+            let typo = iter.next().expect("typo");
+            (
+                typo.to_owned(),
+                iter.map(ToOwned::to_owned).collect::<Vec<_>>(),
+            )
         });
+    dict_from_iter(iter)
+}
 
-    let rows: Dict = rows
+fn dict_from_iter<S: Into<String>>(
```

Comment:
I think I'd prefer `parse` to return a `Vec` and for `process` to accept it.

Reply:
Good call, this also lets us easily test that duplicate entries are merged ... I made that change and added that test :)